client/conn.go

```diff
@@ -45,6 +45,7 @@ type Conn struct {
 	capability uint32
 	// client-set capabilities only
 	ccaps uint32
+	dcaps uint32 // disabled capabilities
```

Comment:
I see you followed the naming style of `ccaps`. Just a weak comment: `capability`, `ccaps` and `dcaps` are hard to understand at their usage functions. Maybe rename them to `serverCaps`, `clientExplicitOnCaps`, `clientExplicitOffCaps` or something.